components/brave_wallet_ui/page/screens/composer_ui/select_token_modal/select_token_modal.tsx

```diff
@@ -167,270 +235,296 @@ export const SelectTokenModal = React.forwardRef<HTMLDivElement, Props>(
+      )
+    }, [networks, selectedAccountFilter, tokensFilteredByNetwork])
+
+    const tokensBySearchValue = React.useMemo(() => {
+      if (searchValue === '') {
+        return tokensFilteredByAccount
+      }
+      const lowercaseSearchValue = searchValue.toLowerCase()
+      return tokensFilteredByAccount.filter(
+        (token) =>
+          token.asset.name.toLowerCase() === lowercaseSearchValue ||
+          token.asset.name.toLowerCase().startsWith(lowercaseSearchValue) ||
+          token.asset.symbol.toLocaleLowerCase() === lowercaseSearchValue ||
+          token.asset.symbol.toLowerCase().startsWith(lowercaseSearchValue) ||
+          token.asset.contractAddress.toLocaleLowerCase() ===
+            lowercaseSearchValue
+      )
+    }, [tokensFilteredByAccount, searchValue])
+
+    const firstNoBalanceTokenKey = React.useMemo(() => {
+      const token = tokensBySearchValue.find((token) =>
+        new Amount(token.assetBalance).isZero()
+      )
+      return token ? getAssetIdKey(token.asset) : ''
+    }, [tokensBySearchValue])
+
+    const accountsForPendingSelectedAsset = React.useMemo(() => {
+      if (!pendingSelectedAsset) {
+        return []
+      }
+      return accounts
+        .filter(
+          (account) => account.accountId.coin === pendingSelectedAsset.coin
+        )
+        .sort(function (a, b) {
+          return new Amount(
+            getBalance(b.accountId, pendingSelectedAsset, tokenBalancesRegistry)
+          )
+            .minus(
+              getBalance(
+                a.accountId,
+                pendingSelectedAsset,
+                tokenBalancesRegistry
+              )
+            )
+            .toNumber()
         })
-        return new Amount(reducedAmounts).formatAsFiat(defaultFiatCurrency)
-      },
-      [
-        getTokensBySearchValue,
-        spotPriceRegistry,
-        defaultFiatCurrency,
-        tokenBalancesRegistry
-      ]
-    )
+    }, [accounts, pendingSelectedAsset, tokenBalancesRegistry])
 
+    // Methods
     const handleSelectAsset = React.useCallback(
-      (
-        token: BraveWallet.BlockchainToken,
-        account?: BraveWallet.AccountInfo
-      ) => {
-        onSelectAsset(token, account)
-        onClose()
+      (token: BraveWallet.BlockchainToken) => {
+        // No need to select an account when selecting
+        // a token to receive right now.
+        // This will change with bridge.
+        if (selectingFromOrTo === 'to') {
+          onSelectAsset(token)
+          onClose()
+          return
+        }
+
+        setPendingSelectedAsset(token)
       },
-      [onSelectAsset, onClose]
+      [onSelectAsset, onClose, selectingFromOrTo]
     )
 
-    const onSearch = React.useCallback(
-      (event: React.ChangeEvent<HTMLInputElement>) => {
-        setSearchValue(event.target.value)
+    const handleSelectTokenDetails = React.useCallback(
+      (token: BraveWallet.BlockchainToken) => {
+        setTokenDetails(token)
```

Comment:
This looks like it is just a wrapper for `setTokenDetails`. Can this be removed?

Reply:
Removed and passed directly!